### ISSUE TITLE
EIA_API_KEY in GitHub Actions

### ISSUE THREAD

**Issue by WillKoehrsen** [kmax12/gridstatus#310]
The `EIA_API_KEY` is blank in the GitHub actions environment, causing four of the `test_eia` tests to fail. 

This can be recreated locally:

```bash
export EIA_API_KEY=

pytest gridstatus/tests/test_eia.py

FAILED gridstatus/tests/test_eia.py::test_list_routes - KeyError: 'response'
FAILED gridstatus/tests/test_eia.py::test_rto_interchange - KeyError: 'response'
FAILED gridstatus/tests/test_eia.py::test_rto_region_data - KeyError: 'response'
FAILED gridstatus/tests/test_eia.py::test_fuel_type - KeyError: 'response'
```

Setting a valid value for `EIA_API_KEY` allows the build to pass locally.

Secrets are not shared with PRs from forked repos unless we use the [`pull_request_target` trigger](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target) in GitHub actions (I think based on an [explanation](https://stackoverflow.com/a/74959635/5755357). So, my recommendation would be to change the `pull_request` to `pull_request_target` in `./.github/workflows/tests.yaml`.

**Comment by WillKoehrsen** [comment]
Or, if the key doesn't need to be kept secret we could set it as a repo variable. I don't think that's a high security risk since we aren't paying for the key.

**Comment by kmax12** [comment]
we can make it a repo variable

**Comment by WillKoehrsen** [comment]
Addressed in #321 by putting the API key directly into the GitHub actions file since the key does not need to be secret.
